js/cards/smartport.ts

```diff
@@ -0,0 +1,460 @@
+    }
+
+    readWord() {
+        const readLo = this.readByte();
+        const readHi = this.inc(1).readByte();
+
+        return readHi << 8 | readLo;
+    }
+
+    readAddress() {
+        const readLo = this.readByte();
+        const readHi = this.inc(1).readByte();
+
+        return new Address(this.cpu, readLo, readHi);
+    }
+
+    writeByte(val: byte) {
+        this.cpu.write(this.hi, this.lo, val);
+    }
+
+    writeWord(val: word) {
+        this.writeByte(val & 0xff);
+        this.inc(1).writeByte(val >> 8);
+    }
+
+    writeAddress(val: Address) {
+        this.writeByte(val.loByte());
+        this.inc(1).writeByte(val.hiByte());
+    }
+
+    toString() {
+        return '$' + toHex(this.hi) + toHex(this.lo);
+    }
+}
+
+export default class SmartPort implements Card, Restorable<SmartPortState> {
+    COMMAND = 0x42;
+    UNIT = 0x43;
+    ADDRESS_LO = 0x44;
+    // const ADDRESS_HI = 0x45;
+    BLOCK_LO = 0x46;
+    // const BLOCK_HI = 0x47;
+
+    private rom: Uint8Array;
+    private disks: SmartDisk[] = [];
+
+    constructor(private cpu: CPU6502, options: SmartPortOptions) {
+        this.rom = new Uint8Array(rom);
+        if (options?.block) {
+            this.rom[0x07] = 0x3C;
+            debug('DumbPort card');
+        } else {
+            debug('SmartPort card');
+        }
+    }
+
+    private decodeDisk(unit: number, disk: BlockDevice) {
+        this.disks[unit] = [];
+        for (let idx = 0; idx < disk.blocks.length; idx++) {
+            this.disks[unit][idx] = new Uint8Array(disk.blocks[idx]);
+        }
+    }
+
+    private debug(..._args: any[]) {
+        // debug.apply(this, arguments);
+    }
+
+
+    /*
+     * dumpBlock
+     */
+    /*
+    dumpBlock(drive, block) {
+        const result = '';
+        const b;
+        const jdx;
+
+        for (const idx = 0; idx < 32; idx++) {
+            result += toHex(idx << 4, 4) + ': ';
+            for (jdx = 0; jdx < 16; jdx++) {
+                b = disks[drive][block][idx * 16 + jdx];
+                if (jdx == 8) {
+                    result += ' ';
+                }
+                result += toHex(b) + ' ';
+            }
+            result += '        ';
+            for (jdx = 0; jdx < 16; jdx++) {
+                b = disks[drive][block][idx * 16 + jdx] & 0x7f;
+                if (jdx == 8) {
+                    result += ' ';
+                }
+                if (b >= 0x20 && b < 0x7f) {
+                    result += String.fromCharCode(b);
+                } else {
+                    result += '.';
+                }
+            }
+            result += '\n';
+        }
+        return result;
+    }
+*/
+    /*
+     * getDeviceInfo
+     */
+
+    getDeviceInfo(state: CpuState, drive: number) {
+        if (this.disks[drive]) {
+            const blocks = this.disks[drive].length;
+            state.x = blocks & 0xff;
+            state.y = blocks >> 8;
+
+            state.a = 0;
+            state.s &= 0xfe;
```

Comment:
How about `state.s &= ~flags.C`? It makes it clearer that this is clearing the carry flag. Same below in `readBlock`, `writeBlock`, etc.